File: ebook/src/main/java/br/com/ebook/foobnix/sys/TempHolder.java
package br.com.ebook.foobnix.sys;

import org.ebookdroid.BookType;

import java.util.concurrent.locks.ReentrantLock;

import br.com.ebook.foobnix.pdf.info.ExtUtils;

public class TempHolder {
    public static final ReentrantLock lock = new ReentrantLock();
    public static TempHolder inst = new TempHolder();

    public volatile String path;
    public boolean isTextFormat;
    public boolean isTextForamtButNotTxt;

    public String login = "", password = "";

    public int linkPage = -1;

    public static int listHash = 0;

    public static volatile boolean isSeaching = false;
    public static volatile boolean isConverting = false;
    public static volatile boolean isRecordTTS = false;

    public long timerFinishTime = 0;

    public int pageDelta = 0;

    public volatile boolean loadingCancelled = false;
    public boolean forseAppLang = false;

    public volatile long lastRecycledDocument = 0;

    public static TempHolder get() {
        return inst;
    }

    public void init(String pathI) {
        path = pathI;
        isTextFormat = isTextForamtInner();
        isTextForamtButNotTxt = isTextForamtButNotTxt();
    }

    public void clear() {
        path = null;
    }

    private boolean isTextForamtInner() {
        try {
            return ExtUtils.isTextFomat(path);
        } catch (Exception e) {
            return false;
        }
    }

    private boolean isTextForamtButNotTxt() {
        try {
            return ExtUtils.isTextFomat(path) && !BookType.TXT.is(path);
        } catch (Exception e) {
            return false;
        }
    }

}

File: ebook/src/main/java/br/com/ebook/foobnix/sys/TempHolder.kt
package br.com.ebook.foobnix.sys

import br.com.ebook.foobnix.pdf.info.ExtUtils
import org.ebookdroid.BookType
import java.util.concurrent.locks.ReentrantLock

class TempHolder {

    @JvmField
    @Volatile
    var path: String? = null
    
    @JvmField
    var isTextFormat: Boolean = false
    
    @JvmField
    var isTextFormatButNotTxt: Boolean = false

    @JvmField
    var login = ""
    
    @JvmField
    var password = ""
    
    @JvmField
    var linkPage = -1
    
    @JvmField
    var timerFinishTime = 0L
    
    @JvmField
    var pageDelta = 0

    @JvmField
    @Volatile
    var loadingCancelled = false
    
    @JvmField
    var forceAppLang = false

    @JvmField
    @Volatile
    var lastRecycledDocument = 0L

    fun init(pathI: String) {
        path = pathI
        isTextFormat = isTextFormatInner()
        isTextFormatButNotTxt = checkIsTextFormatButNotTxt()
    }

    fun clear() {
        path = null
    }

    private fun isTextFormatInner(): Boolean {
        return try {
            ExtUtils.isTextFomat(path)
        } catch (e: Exception) {
            false
        }
    }

    private fun checkIsTextFormatButNotTxt(): Boolean {
        return try {
            ExtUtils.isTextFomat(path) && !BookType.TXT.`is`(path)
        } catch (e: Exception) {
            false
        }
    }

    companion object {
        @JvmField
        val lock = ReentrantLock()
        
        @JvmField
        val inst = TempHolder()

        @JvmStatic
        fun get(): TempHolder = inst

        @JvmField
        var listHash = 0

        @Volatile
        @JvmField
        var isSearching = false

        @Volatile
        @JvmField
        var isConverting = false

        @Volatile
        @JvmField
        var isRecordTTS = false
    }
}
